frontend/scripts/setup/mac_setup.sh

```
@@ -0,0 +1,303 @@
+        _java=java
+    elif [[ -n "$JAVA_HOME" ]] && [[ -x "$JAVA_HOME/bin/java" ]]; then
+        _java="$JAVA_HOME/bin/java"
+    else
+        echo_error "Java is not installed"
+        return 1
+    fi
+
+    java_ver=$("$_java" -version 2>&1 | awk -F '"' '/version/ {print $2}')
+    echo_info "Current Java version: $java_ver"
+    
+    if [[ "$java_ver" < "17" ]]; then
```

Shellcheck (0.10.0)

[error] < is for string comparisons. Use -lt instead.

(SC2071)